### PR TITLE
refactor: make ELECTRON_INSPECTOR_CONTEXT_MENU handler async

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
Simplifies the flow of handling the context menu

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes
Notes: no-notes

### CODE REVIEW

**Comment by miniak** [comment]
@nornagon can you please have a look?

**Comment by miniak** [comment]
@nornagon can you please check?

File: lib/browser/chrome-devtools.js [hunk 1]
```diff
@@ -67,22 +65,22 @@ const assertChromeDevTools = function (contents, api) {
...
+
+    const template = isEditMenu ? getEditMenuItems() : convertToMenuTemplate(items, resolve)
+    const menu = Menu.buildFromTemplate(template)
+    const window = event.sender.getOwnerBrowserWindow()
+
+    menu.once('menu-will-close', () => {
+      // menu-will-close is emitted before the click handler, but needs to be sent after.
+      // otherwise, DevToolsAPI.contextMenuCleared() would be called before
+      // DevToolsAPI.contextMenuItemSelected(id) and the menu will not work properly.
+      setTimeout(() => resolve())
```
Comment (nornagon) [inline lib/browser/chrome-devtools.js]:
```suggestion
      setTimeout(resolve)
```

Comment (nornagon) [reply lib/browser/chrome-devtools.js]: eh nvm :)

**Pull request merged**